src/test/java/com/epam/tc/nitcenkov/hw4/tests/IndexPageTest.java

```diff
@@ -1,64 +1,64 @@
 package com.epam.tc.nitcenkov.hw4.tests;
 
-import com.epam.tc.nitcenkov.hw4.components.HeaderComponent;
-import com.epam.tc.nitcenkov.hw4.components.LeftSectionComponent;
-import com.epam.tc.nitcenkov.hw4.components.LoginComponent;
 import com.epam.tc.nitcenkov.hw4.listener.ScreenshotListener;
 import com.epam.tc.nitcenkov.hw4.pages.IndexPage;
-import com.epam.tc.nitcenkov.hw4.util.Util;
+import com.epam.tc.nitcenkov.hw4.util.DataProviderClass;
 import io.qameta.allure.Epic;
 import io.qameta.allure.Feature;
 import io.qameta.allure.Severity;
 import io.qameta.allure.SeverityLevel;
 import io.qameta.allure.Story;
-import java.util.Arrays;
 import java.util.List;
 import org.assertj.core.api.SoftAssertions;
-import org.openqa.selenium.support.PageFactory;
 import org.testng.annotations.Listeners;
 import org.testng.annotations.Test;
 
 @Listeners(ScreenshotListener.class)
 public class IndexPageTest extends BaseTest {
 
-    SoftAssertions softAssertions = new SoftAssertions();
+    private static final int EXPECTED_IMAGES_NUMBER = 4;
+    private static final int EXPECTED_TEXTS_NUMBER = 4;
+    private static final int EXPECTED_HEADER_BUTTONS_NUMBER = 4;
```

Comment:
Would be better to replace hardcoded values with something like `expectedIndexPageTexts.size()`